packages/utilities/src/utilities/StickyHeader/StickyHeader.js

```diff
@@ -171,116 +186,207 @@ class StickyHeader {
+   */
+  _calculateMaxScrollaway() {
+    const {
+      masthead,
+      mastheadL0,
+      mastheadL1,
+      tableOfContents,
+      tableOfContentsInnerBar,
+      leadspaceSearchBar,
+    } = StickyHeader.global._elements;
+
+    // Reset the value before performing any further calculations.
+    this._state.maxScrollaway = 0;
+
+    // Collect conditions we may want to test for to make logic easier to read.
+    const tocShouldStick = tableOfContents
+      ? tableOfContents.layout === 'horizontal' ||
+        window.innerWidth < gridBreakpoint
+      : false;
+    const tocIsAtTop = tableOfContentsInnerBar
+      ? tableOfContentsInnerBar.getBoundingClientRect().top <=
+        (masthead ? masthead.offsetTop + masthead.offsetHeight : 0) + 1
+      : false;
+    const searchIsAtTop = leadspaceSearchBar
+      ? leadspaceSearchBar.getBoundingClientRect().top <=
+        (masthead ? masthead.offsetTop + masthead.offsetHeight : 0) + 1
+      : false;
+    const tocIsAtSearch =
+      leadspaceSearchBar && tableOfContentsInnerBar
+        ? tableOfContentsInnerBar.getBoundingClientRect().top <=
+          leadspaceSearchBar.getBoundingClientRect().bottom
+        : false;
+    const mastheadL1IsActive = mastheadL1 && mastheadL1.hasAttribute('active');
+
+    // Begin calculating maxScrollAway.
+
+    // If L1 is open, lock it to the top of the page.
+    if (mastheadL1IsActive && mastheadL0) {
+      this._state.maxScrollaway = mastheadL0.offsetHeight;
+    } else {
+      // In cases where we have both an eligible ToC and leadspace search, we want
+      // the ToC to take precedence. Scroll away leadspace search.
+      if (tocIsAtSearch && tocShouldStick) {
+        this._state.maxScrollaway += leadspaceSearchBar.offsetHeight;
+      }
 
-      if (masthead && searchIsAtTop) {
-        maxScrollaway += masthead.offsetHeight;
+      // Scroll away entire masthead if either ToC or leadspace search is eligible
+      // to be the stuck element (unless L1 is open). Otherwise, scroll away the
+      // L0 if we have an L1.
+      if (searchIsAtTop || (tocIsAtTop && tocShouldStick)) {
+        if (masthead) {
+          this._state.maxScrollaway += masthead.offsetHeight;
+        }
+      } else if (masthead && mastheadL0 && mastheadL1) {
+        this._state.maxScrollaway += mastheadL0.offsetHeight;
       }
     }
+  }
+
+  /**
+   * Positions sticky elements. Does so by checking the scroll position and where
+   * tracked elements are in relation to it, then applying the correct styles to
+   * each element in succession to ensure that only one element is stuck to the
+   * top of the page, and all other elements that have been scrolled past can be
+   * revealed when scrolling back up.
+   */
+  _positionElements() {
+    const {
+      banner,
+      masthead,
+      tableOfContentsInnerBar: tocInner,
+      leadspaceSearchBar,
+    } = StickyHeader.global._elements;
+    const { scrollPosPrevious: oldY } = StickyHeader.global._state;
```

Comment:
Same note here as in `_manageStickyElements`. Since this method is called exclusively from `_manageStickyElements`, should be safe by the same rationale to use `this`.